### DISCUSSION TITLE
Editor enhancement options

### DISCUSSION THREAD

**Discussion by austintoddj** [austintoddj/canvas#919] [New Features]
[This issue](https://github.com/austintoddj/canvas/issues/837) sparked some conversation about various features that could make the editor more useful. Since it's not really an issue with the project, I'm opening the discussion here to brainstorm more solutions that can be brought into a future release.

----

While [Quill.js](https://quilljs.com/) is a powerful tool with great editing capabilities, it'd be helpful to have other options when it comes to an editor. Some of the viable packages we could include here are:

- [TipTap](https://tiptap.dev)
- [Dante](https://michelson.github.io/dante2)

I'm sure there are more. This at least gets us started on ways to improve the editing experience with Canvas. 

## Feature requests
- Specify the size/dimensions or position (left align, center, right align, fixed etc) of an image
- [Liquid tags](https://github.com/austintoddj/canvas/issues/541)
- Markdown support
- "Link-to" sections within a post
- Scientific notation support
- HTML5 video support
- Link customization (follow, nofollow, specify targets)

**Comment by usamapatel** [comment]
Can we add text alignment options to text editor like we have in ckeditor and other rich text editors?

**Reply by austintoddj** [reply]
The plan for v7 which is to use [TipTap](https://tiptap.dev/).  There'll be a lot more options available to us (like alignment) once that's in place.

**Comment by febrihidayan** [comment]
I personally really like the editor from Medium, of course, what is in Canvas. However, it is better to add video features from YouTube, Facebook, Instagram, and other social media. I totally agree with this point

**Comment by ghost** [comment]
I am not a programmer or anything like that. I want to talk from user perspective. It would be nice to have an option in the user interface to change editor type. I think an editor like "laraberg" will be much appreciated. It is like Gutenberg for laravel. www.github.com/vanOns/laraberg

**Comment by ghost** [comment]
Woah! Will be glad. I salute you for the good works!

**Comment by ghost** [comment]
Is there a definite time frame for v7 yet?

**Reply by austintoddj** [reply]
There's not. Due to life circumstances and such, it's taken far longer than anticipated. But it's cruising along now, so keep your [eyes on the releases](https://github.com/austintoddj/canvas/releases)!

**Comment by tvarga94** [comment]
@austintoddj for now, do we have a workaround to at least center a text?

Also couple of things that would be quite important for me, but could not figure out how to do it:
1. Can we change text color?
2. I can not even copy a table in the post. Is there way to handle table structures?

Thank you for the answers in advance!

**Comment by zusamarehan** [comment]
@austintoddj I can maintain this project for you, if you are interested.

**Comment by austintoddj** [comment]
Closing this out with the [Canvas v7.0.0](https://github.com/austintoddj/canvas/releases/tag/v7.0.0) release.

The admin editor is now [TipTap](https://tiptap.dev/) (not Quill), with alignment, code blocks (with language), YouTube embeds, images, tables, and more of the capabilities discussed here.

If something specific is still missing on v7, please open a new discussion or issue against current `main`.
